File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0;
        int r = 1;
        int max = 0;

        while (r < prices.size()) {
            if (prices[r] < prices[l]) {
                l = r;
            } else if (prices[r] - prices[l] > max){
                max = prices[r] - prices[l];
            }
            r++;
        }
        return max;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]                    # init maxSubArray var to 1st elem in nums
        curSum = 0                          # init curSum var to 0

        for n in nums:                      # traverse each num in nums:
            if curSum < 0:                      # if curSum after prev loop is negative:
                curSum = 0                          # reset curSum to 0
            curSum += n                         # add cur num to curSum
            maxSub = max(maxSub, curSum)        # update maxSub to max of cur vals and curSum

        return maxSub                       # return maxSub